flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/sink/dynamic/DynamicCommitter.java

```diff
@@ -210,36 +218,40 @@ private void commitPendingRequests(
     NavigableMap<Long, List<WriteResult>> pendingResults = Maps.newTreeMap();
     for (Map.Entry<Long, List<CommitRequest<DynamicCommittable>>> e : commitRequestMap.entrySet()) {
       for (CommitRequest<DynamicCommittable> committable : e.getValue()) {
-        if (Arrays.equals(EMPTY_MANIFEST_DATA, committable.getCommittable().manifest())) {
+        if (Arrays.deepEquals(EMPTY_MANIFEST_DATA, committable.getCommittable().manifests())) {
```

Comment:
This logic is not good in this way. We need to combine it to the `manifests` loop below.

The end goal what we want to archive:
- Every manifest should be in the `pendingResults` which is not `EMPTY_MANIFEST_DATA`
- If the `pendingResults` is empty, then put a single `EMPTY_MANIFEST_DATA` into it

We shouldn't put `EMPTY_MANIFEST_DATA` to the `pendingResults`. And but put